[READY]Addition of new explosive chems

Djiq opened the pull request:
## About The Pull Request
Addition of 8 new chems. 4 of which are used to craft the other 4. Mainly explosives .Nerfed water potassium, because it is too easy to create insta-gib grenades. Addition of a single explosive that is at the same time a medical substance

## Why it is good for the game
Creating insta-gib grenades is too easy, as all you needed were 2 clicks, which is something bad as explosions that can gib shouldnt be available neigh roundstart. I added 3 new explosives : TaTP , RDX and Penthrite. TaTP is now the second most powerful explosive. handling it is very finnicky as the explosion and reaction temperature changes with every round. RDX is an easier and more controlled substance, tho not as powerful as TaTP.Additionally RDX has some special explosive interactions with Liquid Electricity and Teslium. Penthrite is a medical chem used for stabilizing hearts. Simmiliary to Corazone it stops heart attacks. When mixed with phenol and acetone oxide(new chem) and heated to 315k it explodes. Creating all of these chems is very annoying , and has low error margin, making them high risk , high reward explosives. Added Nitric Acid. a highly corrosive substance that quickly corrodes living tissue.


## Changelog
:cl:Edge
add: Added new toxin: Nitric Acid 
add: Added new explosive: TaTP <-have fun with that one
add: Added new explosive: RDX , don't mix it with electricity!
add: Added new explosive/medical : Penthrite 
add: Added chem: Hydrogen peroxide
add: Added chem: Acetone oxide
add: Added chem: Acetaldehyde
add: Added chem: Pentaerythritol
balance: Removed the heart stabilizing effect of Corazone, look at Penthrite for that effect
balance: Nerfed Water+Potassium explosion so it is harder to gib someone 
/:cl:

Djiq committed:
boomies

Djiq committed:
updated some recipes

bobbahbrown commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,21 @@
+/datum/reagent/medicine/penthrite
+	// Same as corazone. but slightly better
+	name = "penthrite"
+	description = "A medication used to treat certain heart conditions, also it is explosive"
```
Awkward wording, perhaps "A medication used to treat certain heart conditions, it is also explosive." or "An explosive compound used to treat certain heart conditions." etc

bobbahbrown replied:
```suggestion
	description = "An explosive compound used to treat certain heart conditions."
```

bobbahbrown commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -251,6 +251,54 @@
 
+/datum/reagent/hydrogen_peroxide
+	name = "Hydrogen peroxide"
+	description = "An ubiquitous chemical substance that is composed of hydrogen and oxygen and oxygen."
```
Extra "and oxygen"

bobbahbrown commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,21 @@
 	..()
+/datum/reagent/medicine/penthrite
+	// Same as corazone. but slightly better
+	name = "penthrite"
```
Name should start with an uppercase to follow the existing convention

bobbahbrown replied:
```suggestion
	name = "Penthrite"
```

bobbahbrown commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -251,6 +251,54 @@
+	var/cooling_temperature = 2
+	glass_icon_state = "glass_clear"
+	glass_name = "glass of oxygenated water"
+	glass_desc = "The father of all refreshments. surely it tastes great? right.."
```
Grammatical error. "The father of all refreshments. Surely it tastes great, right?" or "The father of all refreshments. Surely it tastes great? Right..." -- note ellipsis (...) have three periods if you decide to keep it.

bobbahbrown replied:
```suggestion
	glass_desc = "The father of all refreshments. Surely it tastes great, right?"
```

bobbahbrown commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1480,6 +1528,34 @@
 
+/datum/reagent/hexamine
+	name = "Hexamine"
+	description = "A White crystalline compound useful for synthesizing other chemicals"
```
Grammatical error -- "A white crystalline compound useful for synthesizing other chemicals."

bobbahbrown replied:
```suggestion
	description = "A white crystalline compound useful for synthesizing other chemicals."
```

bobbahbrown commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1480,6 +1528,34 @@
+
+/datum/reagent/pentaerythritol
+	name = "Pentaerythritol"
+	description = "Slow down , it aint no spelling bee!"
```
Typing error, extra space after "Slow down" -- also "ain't" should have an apostrophe.

bobbahbrown replied:
```suggestion
	description = "Slow down, it ain't no spelling bee!"
```

bobbahbrown commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1480,6 +1528,34 @@
+
+/datum/reagent/acetaldehyde
+	name = "Acetaldehyde"
+	description = "Simmiliar to plastic. Tastes like dead people"
```
"Similar"

bobbahbrown replied:
```suggestion
	description = "Similar to plastic. Tastes like dead people."
```

bobbahbrown commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -42,14 +42,72 @@
+
+/datum/chemical_reaction/reagent_explosion/tatp/New()
+	..()
+	required_temp = 450 + rand(-49,49)
```
This doesn't change every round, this changes with _every call_

bobbahbrown commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -251,6 +251,54 @@
 
+/datum/reagent/hydrogen_peroxide
+	name = "Hydrogen peroxide"
+	description = "An ubiquitous chemical substance that is composed of hydrogen and oxygen and oxygen."
```
```suggestion
	description = "An ubiquitous chemical substance that is composed of hydrogen and oxygen."
```

EdgeLordExe replied:
this is actually on purpose, it is a pun

bobbahbrown replied:
hmm!

Djiq committed:
Update code/modules/reagents/chemistry/reagents/medicine_reagents.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

Djiq committed:
Update code/modules/reagents/chemistry/reagents/medicine_reagents.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

Djiq committed:
Update code/modules/reagents/chemistry/reagents/other_reagents.dm

All of these are just grammar fixes

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

Djiq committed:
Update code/modules/reagents/chemistry/reagents/other_reagents.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

Djiq committed:
Update code/modules/reagents/chemistry/reagents/other_reagents.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

Djiq committed:
Update code/modules/reagents/chemistry/reagents/other_reagents.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

Djiq reviewed (commented):
yep all of these are good description changes. i approve heh

yoyobatty commented:
Hexamine, is that by chance inspired from HippieStation's lcass chem of the same name? It's a pretty fun way of setting dudes on fire.

CRITAWAKETS commented:
These all look like amazing chems to mess around with. I do know of how you could get TaTP to trigger an assured explosion every time though. A similar method to actual meth explosives. Since it requires effort and knowledge to do though, im on board with this.

Krysonism commented:
Some of the recipes makes little sense. why are acetone peroxide and its trimer different reagents?

Splashing hydrogen peroxide on someone shouldn't remove fire stacks, if they are covered  in solvent it makes more sense to ignite the mob.

EdgeLordExe commented:
well then if you think that something should be change. start a suggestion

Tlaltecuhtli commented:
remove corazone and use the new chem that does the same thing?

EdgeLordExe commented:
well corazone also works on the liver. this chem only works on hearts

EdgeLordExe commented:
> remove corazone and use the new chem that does the same thing?

corazone also works on livers. penthrite only works on hearts

Djiq committed:
Changes to the behaviour of TaTP, hydrogen peroxide and acetone oxide, changed acetone peroxide to acetone oxide.

Djiq committed:
forgot to // the debug code i used

Djiq committed:
further nerfs to potassium and water explosion. Inital nerf wasnt enough. it still had a gibbing potential

nemvar commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1480,6 +1512,45 @@
+	if(!istype(M))
+		return
+	if(method == TOUCH)
+		M.adjustBruteLoss(2, 0) // burns
```
If it's burns, why is this brute loss?

EdgeLordExe replied:
uuuuuh fuck ill fix it

nemvar commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -244,13 +244,45 @@
+	if(!istype(M))
+		return
+	if(method == TOUCH)
+		M.adjustBruteLoss(2, 0) // burns
```
Second variable is a boolean.

ExcessiveUseOfCobblestone replied:
this wasn't resolved...

nemvar commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -244,13 +244,45 @@
+		T.MakeSlippery(TURF_WET_WATER, 10 SECONDS, min(reac_volume*1.5 SECONDS, 60 SECONDS))
+	var/obj/effect/acid/A = (locate(/obj/effect/acid) in T)
+	if(A)
+		A.acid_level = max(A.acid_level + reac_volume*50, 0)
```
Extremely rare effect that basically doesn't even do anything even if it does trigger. Why does this even exist?

nemvar commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -803,6 +803,18 @@
+	acidpwr = 5.0
+
+/datum/reagent/toxin/acid/nitracid/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(current_cycle/12, 0)
```
Second argument is a bool.

EdgeLordExe replied:
> What exactly is the point of adding multiple new chems that either don't do anything or do basically the same thing?
> Why exactly do we need more murder chems?
> Please atomize this.

variety mainly. Also because i wanted to nerf these extremely easy to create explosives. These are more powerful but also very finnicky to deal with.

nemvar commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -803,6 +803,18 @@
+
+/datum/reagent/toxin/acid/nitracid/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(current_cycle/12, 0)
+	. = 1
```
. = TRUE

nemvar commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -42,14 +42,82 @@
+	SSticker.OnRoundstart(CALLBACK(src,.proc/UpdateInfo))
+
+/datum/chemical_reaction/reagent_explosion/tatp_explosion/New()
+	SSticker.OnRoundstart(CALLBACK(src,.proc/UpdateInfo))
```
Why is this over SSticker? Just do this the normal way without any callbacks or other fuckery.

EdgeLordExe replied:
well i m new to byond, and i wanted UpdateInfo() to only fire once, at roundstart

EdgeLordExe replied:
and after looking through Secret recipe. and asking a maintainer they said it could be done by ssticker. so i did it

EdgeLordExe replied:
> Why is this over SSticker? Just do this the normal way without any callbacks or other fuckery.

if you have a better solution , please create a suggestion
coz i dont know of a better solution , as this code is also used in secret sauce

nemvar commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -42,14 +42,82 @@
+	strengthdiv = 3
+
+/datum/chemical_reaction/reagent_explosion/tatp/proc/UpdateInfo()
+	..()
```
Parent call doesn't do anything.

nemvar commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -42,14 +42,82 @@
+/datum/chemical_reaction/reagent_explosion/tatp/on_reaction(datum/reagents/holder, created_volume)
+	if(holder.has_reagent(/datum/reagent/stabilizing_agent))
+		return
+	holder.remove_reagent(/datum/reagent/tatp, created_volume*1)
```
``created_volume*1`` lol

EdgeLordExe replied:
i wrote it at 1 am yesterday? today?

EdgeLordExe replied:
ill fix it dont ya worry

nemvar reviewed (commented):
What exactly is the point of adding multiple new chems that either don't do anything or do basically the same thing?
Why exactly do we need more murder chems?
Please atomize this.

Djiq committed:
fixed an error

EdgeLordExe commented:
> What exactly is the point of adding multiple new chems that either don't do anything or do basically the same thing?
> Why exactly do we need more murder chems?
> Please atomize this.

Variety, and because i wanted to nerf water - pot nades. I nerfed them and instead of water-pot you now have harder to make - more destructive chems

Djiq committed:
fixes shit pointed by nemvar owo. Also deleted hexamine, it is replaced by phenol in all of the recipes

carlarctg commented:
the changelog is incredibly useless by the way, make it actually say which chems are inert, which explosives heal heart damage, and which explosive is random

wesoda25 commented:
PATP seems the only worthwhile addition from this PR. The others are just the same old boring explosives that we already have, which is lame. I'm all for adding complexity and interesting mechanics to chemistry pyrotechnics but PATP is the only one that does that. 

Also, nitroglycerin is rarely used because it requires inter department cooperation. Its an insanely strong explosive and I think you can still get near max cap level with it? I don't think it deserves a buff.

And what does potass water do now? 10 to 25 seems a very significant nerf, say in a 50u 50u reaction, does it even breach? 

If you want to prevent gibbing with potass water, I think an interesting thing would be to have them generate a strong shock wave, but not with that much destructive power. So if it goes off it will send you flying and blind/deafen you, but nothing will be breached and you will only be at yellow health. Would definitely make it more interesting and fun, while removing gibbing capacity.

EdgeLordExe commented:
> PATP seems the only worthwhile addition from this PR. The others are just the same old boring explosives that we already have, which is lame. I'm all for adding complexity and interesting mechanics to chemistry pyrotechnics but PATP is the only one that does that.
> 
> Also, nitroglycerin is rarely used because it requires inter department cooperation. Its an insanely strong explosive and I think you can still get near max cap level with it? I don't think it deserves a buff.
> 
> And what does potass water do now? 10 to 25 seems a very significant nerf, say in a 50u 50u reaction, does it even breach?
> 
> If you want to prevent gibbing with potass water, I think an interesting thing would be to have them generate a strong shock wave, but not with that much destructive power. So if it goes off it will send you flying, blind/deafen, and knock you down, but nothing will be breached and you will only be at yellow health. Would definitely make it more interesting and fun, while removing gibbing capability.
hell 100u 100u no longer breaches. I agree i went a little overboard. my main intention was making roundstart gibbing nades impossible to do with 2 clicks. Penthrite is a good sabotage chem, and it replaces the water-pot explosion , RDX is just a shitter version of TaTP that i wanted to add so people can create less powerful explosives, but safer to use at the same time.  Nitroglycerin may be too strong , if this get merged, there will be a possibility of nerfing it

EdgeLordExe commented:
> PATP seems the only worthwhile addition from this PR. The others are just the same old boring explosives that we already have, which is lame. I'm all for adding complexity and interesting mechanics to chemistry pyrotechnics but PATP is the only one that does that.
> 
> Also, nitroglycerin is rarely used because it requires inter department cooperation. Its an insanely strong explosive and I think you can still get near max cap level with it? I don't think it deserves a buff.
> 
> And what does potass water do now? 10 to 25 seems a very significant nerf, say in a 50u 50u reaction, does it even breach?
> 
> If you want to prevent gibbing with potass water, I think an interesting thing would be to have them generate a strong shock wave, but not with that much destructive power. So if it goes off it will send you flying, blind/deafen, and knock you down, but nothing will be breached and you will only be at yellow health. Would definitely make it more interesting and fun, while removing gibbing capability.

Well I agree TaTP is definately the most intersting of them. Penthrite is a good sabotage chem , and the explosion of it is a good replacment of the previously gibbing pot + water. RDX is basically tuned down version of TaTP. i might do something with RDX if this gets merged, because irl at least RDX is used as the maint explosive substance in c4. Do you see where im going with ?. Additionaly Nitroglyc is extremely underused, ive never seen anyone use it , so i buffed it

MCterra10 commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,21 @@
+
+/datum/reagent/medicine/penthrite/on_mob_metabolize(mob/living/M)
+	..()
+	M.adjustOrganLoss(ORGAN_SLOT_HEART,-0.15)
```
@optimumtact did mention that there should be no organ healing chems, but at this point so many have snuck through it's impossible to police

optimumtact commented:
no organ healing chems

Djiq committed:
Makes RDX interesting. Explodes uniquely with teslium and liquid electricity.

EdgeLordExe commented:
> no organ healing chems

ill remove this effect then. gimme  a  second

Djiq committed:
oranges said no organ healing chems. so i removed the healing of Penthrite.

EdgeLordExe commented:
> no organ healing chems

@optimumtact i removed the heart healing of Penthrite. Replaced it with stomach damage. :agony:

EdgeLordExe commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,25 @@
+	if(prob(10))
+		M.adjustOrganLoss(ORGAN_SLOT_STOMACH,REM)
+	M.dizziness += 1
+	M.confused += 1
```
@optimumtact here you go , heart healing removed.

Djiq committed:
qucikfix

EdgeLordExe commented:
oh fuck

EdgeLordExe commented:
uuuuuuuuuuuuuh
wait fixed. fuck

EdgeLordExe commented:
i nearly had a heart attack

lolman360 commented:
good job, revert commits

wesoda25 commented:
All you did was resolve bobbahs suggestions, you need to actually commit them.

EdgeLordExe commented:
> All you did was resolve bobbahs suggestions, you need to actually commit them.

what do you mean they are commited

carlarctg commented:
what's the point of penthrite if it's shitty, explosive, corazone

EdgeLordExe commented:
> what's the point of penthrite if it's shitty, explosive, corazone

Look in pyrotechnics.dm penthrite explosion is ment to 'replace' the current easy to create water pot explosion

bobbahbrown commented:
> > All you did was resolve bobbahs suggestions, you need to actually commit them.
> 
> what do you mean they are commited, look in commits

He's right, good catch @wesoda25 -- you undid the changes you accepted in https://github.com/tgstation/tgstation/pull/47536/commits/54912c9630c98d814bf39e92437363c313699a6d

EdgeLordExe commented:
> > > All you did was resolve bobbahs suggestions, you need to actually commit them.
> > 
> > 
> > what do you mean they are commited, look in commits
> 
> He's right, good catch @wesoda25 -- you undid the changes you accepted in [54912c9](https://github.com/tgstation/tgstation/commit/54912c9630c98d814bf39e92437363c313699a6d)

wait really? oh fuck lemme fix that

Djiq committed:
refixes spelling, thanks wesoda and bobbah for pointing it out

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,25 @@
+	self_consuming = TRUE
+
+/datum/reagent/medicine/penthrite/on_mob_metabolize(mob/living/M)
+	..()
```
```suggestion
	. = ..()
```

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,25 @@
+
+/datum/reagent/medicine/penthrite/on_mob_end_metabolize(mob/living/M)
+	REMOVE_TRAIT(M, TRAIT_STABLEHEART, type)
+	..()
```
i think this might be return value specific too

```suggestion
	. = ..()
```

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -151,9 +151,6 @@
 			qdel(hotspot)
-	var/obj/effect/acid/A = (locate(/obj/effect/acid) in T)
-	if(A)
-		A.acid_level = max(A.acid_level - reac_volume*50, 0)
```
why? also undoc'd?

ExcessiveUseOfCobblestone replied:
still haven't figured this out yet.

actioninja replied:
same here

EdgeLordExe replied:
i thought i fixed it , gimme a second

EdgeLordExe replied:
done

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -244,13 +241,45 @@
+		T.MakeSlippery(TURF_WET_WATER, 10 SECONDS, min(reac_volume*1.5 SECONDS, 60 SECONDS))
+	var/obj/effect/acid/A = (locate(/obj/effect/acid) in T)
+	if(A)
+		A.acid_level = max(A.acid_level + reac_volume*50, 0)
```
what is this suppose to do?

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/toxin_reagents.dm:
```diff
@@ -803,6 +803,18 @@
+	acidpwr = 5.0
+
+/datum/reagent/toxin/acid/nitracid/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(current_cycle/12, FALSE) //here you go nervar
```
this will need to be rounded

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -44,12 +44,104 @@
+
+/datum/chemical_reaction/reagent_explosion/tatp_explosion/proc/UpdateInfo()
+	required_temp = 550 + rand(-49,49)
+	message_admins("TaTP explosion mixed temp [required_temp]")
```
fix your file organization

it should be tatp define, tatp procs, tatp_explosion define, tatp_explosion procs.

ExcessiveUseOfCobblestone replied:
actually i'm confusion. what are you trying to do here?

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -44,12 +44,104 @@
+
+/datum/chemical_reaction/reagent_explosion/tatp/proc/UpdateInfo()
+	required_temp = 450 + rand(-49,49)
+	message_admins("TaTP reaction temp [required_temp]")
```
debug message?

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/medicine.dm:
```diff
@@ -244,3 +244,9 @@
 		new /obj/item/stack/medical/suture/medicated(location)
 	return
 
+datum/chemical_reaction/penthrite
```
```suggestion
/datum/chemical_reaction/penthrite
```

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,25 @@
+		M.adjustOrganLoss(ORGAN_SLOT_STOMACH,-REM)
+	M.dizziness += 1
+	M.confused += 1
+	ADD_TRAIT(M, TRAIT_STABLEHEART, type)
```
this can just be added on add, rather than constantly trying to add it per tick

ExcessiveUseOfCobblestone replied:
still an issue

actioninja replied:
you never fixed this.

EdgeLordExe replied:
i really dont know what to do with this. every single other  chem does this in such a way

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,25 @@
 	..()
+/datum/reagent/medicine/penthrite
+	// Same as corazone. but slightly better
+	name = "Penthrite"
```
need to establish a suffix for organ-related meds. I don't care what it is, it just needs to be doc'd somewhere (preferably the C2 file with all the others).

EdgeLordExe replied:
> need to establish a suffix for organ-related meds. I don't care what it is, it just needs to be doc'd somewhere (preferably the C2 file with all the others).

aight, tell me when you establish em. for now it will stay as it is.

ExcessiveUseOfCobblestone replied:
`rite` is fine, just update the C2 file to include the prefix

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1057,6 +1058,25 @@
+/datum/reagent/medicine/penthrite/on_mob_metabolize(mob/living/M)
+	..()
+	if(prob(10))
+		M.adjustOrganLoss(ORGAN_SLOT_STOMACH,-REM)
```
is this suppose to heal...

I'd also rather it give low constant damage than rng. Eitherway should be a C2.

EdgeLordExe replied:
> is this suppose to heal...
> 
> I'd also rather it give low constant damage than rng. Eitherway should be a C2 (if it's suppose to heal this is mega cringe).

nah no healing, oranges already told me that. this is bascially shittier corazone, i might add more functionality later. And yeah i agree ill make it very low constant damage

EdgeLordExe replied:
wait fuck i see the problem

Djiq committed:
Apply suggestions from code review

cobby's suggested changes

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

Djiq committed:
Fixes shit pointed out by cobby

EdgeLordExe commented:
i think tavis is not working properly? ive been waiting for some time , and i still dont have an output

EdgeLordExe commented:
@ExcessiveUseOfCobblestone for some reason, travis is being dented right now. Everything you pointed out is fixed tho

EdgeLordExe commented:
travis fixed. every issue fixed i think

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -244,13 +241,42 @@
 
+/datum/reagent/hydrogen_peroxide
+	name = "Hydrogen peroxide"
+	description = "An ubiquitous chemical substance that is composed of hydrogen and oxygen and oxygen."
```
if the typo is intentional, add a comment
```suggestion
	description = "An ubiquitous chemical substance that is composed of hydrogen and oxygen and oxygen." //intended intended
```

ExcessiveUseOfCobblestone commented:
there are still open reviews

tmtmtl30 commented:
the liquid electricity / teslium reactions are a good idea but you need to set the explosion reaction to produce twice as much / the strength divider needs to be twice as low, or else 100u of rdx and 100u of teslium will produce 100u of 4-divider explosion, which is still worse than 200u of rdx making 200u of 6-divider explosion.

at least i think that's how it works

Djiq committed:
Update code/modules/reagents/chemistry/reagents/other_reagents.dm

cobby fix

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

EdgeLordExe commented:
> the liquid electricity / teslium reactions are a good idea but you need to set the explosion reaction to produce twice as much / the strength divider needs to be half its current value, or else 100u of rdx and 100u of teslium will produce 100u of 4-divider explosion, which is still worse than 200u of rdx making 200u of 6-divider explosion.
> 
> at least i think that's how it works

making it half the value , would make rdx absolutely busted in every sense of the way. it shouldnt be dont for the sake of balance. i understand your concerns , but the explosion impact that is currently done is already quite noticable

tmtmtl30 commented:
no, no, i just mean you have to account for the reagents. if you have an explosion with 2 reagents, it'll have half the strength of an equivalent one-reagent explosion. let's say you have a hypothetical heat activated explosion that uses only 1 chem with a strength divider of 10, and another explosion that triggers instantly at any temperature once 3 separate reagents -- at 1 unit each -- come into contact with each other, and this explosion has a strength divider of 8.

if you fill 2 beakers with 100u of the first explosion's reagent and you put them in a chem grenade near their explosion point, all 200u of chems will be heated, and this will result in 200 "units" of explosion, at a strength divider of 10, for an equivalent explosion power of 20.

now, if you fill 2 beakers with 66 units each of the 3 reagents necessary to make the second explosion happen, you end up with 198 units reacting together. since the reaction requires 1 of each chem, there will be enough for 66 "reactions" (though this is actually done all at once), resulting in 66 "units" of explosion. this is then affected by the strength divider of 8, resulting in 8.25, which is then rounded down to an equivalent explosion power of 8. this is, surprisingly, worse than the first reaction.

the best way to demonstrate this is to take a look at the meth explosion code and then test it in a private server. the first meth explosion (/datum/chemical_reaction/reagent_explosion/methsplosion/) has a strength divider of 6, requires one reagent (meth), and explodes at 380 K. the second meth explosion (/datum/chemical_reaction/reagent_explosion/methsplosion/methboom2) is a child of this first explosion. it changes three things, of which one is important: instead of requiring 1 unit of meth, it requires 1 unit _each_ of diethylamine, iodine, phosphorous, and hydrogen. the strength divider is not modified. so we predict that the second meth explosion will be smaller than the first, because in a 200u grenade, the first reaction will result in 200 "units" of explosion, while the second reaction will only result in 50 "units" of explosion. and when we test it, this is exactly what we observe.

so, back to the rdx explosion reactions. the first explosion takes 1 reagent, rdx, and has a strengthdiv of 6, returning an equivalent explosion power of 33 in a 200u grenade. the other explosions take 2 reagents. this means you max out at 100u of each in a 200u grenade. per the above, this means that 100 "units" of explosion will be produced. the liquid electricity + rdx explosion has a strengthdiv of 5; 100 / 5 = an equivalent explosion power of 20. the teslium + rdx explosion has a strengthdiv of 4; 100 / 4 = an equivalent explosion power of 25. with the present code, they will be worse than just straight rdx. presumably, the intention was for the equivalent explosion powers of 200u grenades to look something like 33:40:50. to achieve this, you'd have to halve the strength dividers.

i was, i believe, mistaken on one thing, that being that there is no way to alter the "unit" output of a reagent explosion -- the strength divider has to be set lower to compensate. regardless, _something_ has to be done to account for the lower amount of reactions that are able to take place. sorry for being so verbose

Djiq committed:
Minor change to how rdx works

EdgeLordExe commented:
FUCK

EdgeLordExe commented:
i think i messed something up in git , ffs

EdgeLordExe commented:
i accidentaly set my local master origin to this branch ...fuck

actioninja reviewed (changes requested):
don't mark stuff resolved without actually having made changes

EdgeLordExe commented:
> don't mark stuff resolved without actually having made changes

the fuck? im looking through the files, and both of these are fixed.

Djiq committed:
Fixed shit i thought i already fixed

Djiq committed:
minor error fixed

Djiq committed:
again, minor error on my part fixed

EdgeLordExe commented:
> > don't mark stuff resolved without actually having made changes
> 
> the fuck? im looking through the files, and both of these are fixed.

nevermind , i didnt commit one of these changes. the other one was rightfully pointed out

Djiq committed:
puts penthrite in C2. Removes heart stabilizing effect of corazone.

Djiq committed:
minor addition of annotations

Djiq committed:
minor fix?

Djiq committed:
again minor fix , trying to figure out the merge conflict

Djiq committed:
still figuring it out

Djiq committed:
github forces me to change the description of corazone into a previous state which is inacurate

Djiq committed:
well lets fix it

Djiq committed:
fixes the description

Djiq committed:
Merge branch 'master' into morechems

Djiq committed:
minor fix.. again

EdgeLordExe commented:
i think i have fixed everysingle problem

Djiq committed:
last fix i hope

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -44,12 +44,105 @@
+	var/turf/T = get_turf(holder.my_atom)
+	for(var/turf/turf in range(fire_range,T))
+		new /obj/effect/hotspot(turf)
+	holder.chem_temp = 750
```
this is a copypaste of the one above it, you should modularize `.../rdx_explosion` then subtype

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/pyrotechnics.dm:
```diff
@@ -44,12 +44,105 @@
+	holder.remove_reagent(/datum/reagent/tatp, created_volume)
+	..()
+
+/datum/chemical_reaction/reagent_explosion/tatp_explosion
```
same here

ExcessiveUseOfCobblestone reviewed (commented):
can you remove the comments like `fixed` and `here you go nerv`?

EdgeLordExe commented:
> can you remove the comments like `fixed` and `here you go nerv`?

im on phone currently but sure

ExcessiveUseOfCobblestone commented:
im not going to delete ur pr because you didn't do it within an hour, don't walk n text sweetie!

Djiq committed:
removes unnecessary comment

EdgeLordExe commented:
> im not going to delete ur pr because you didn't do it within an hour, don't walk n text sweetie!

yeah i know cobbwy chan

actioninja commented:
boom boom shake the room

wesoda25 commented:
you did WHAT to corazone?
